### ISSUE TITLE
Implement "Pretranslate", "Copy", "Reset", "Search", "Raw", "Unit #" keyboard shortcuts

### ISSUE THREAD

**Issue by iafan** [translate/pootle#5100]
(See #3136 as a global issue on implementing shortcuts).

Let's implement the following shortcuts (these require minimal programming, so I though it would be easier to fix them all at once):

1. "Pretranslate using MT", mapped to `esc enter`
2. "Reset to current translation", mapped to `esc backspace`
3. "Toggle Raw mode", mapped to `mod mod`.
4. "Search" alternative shortcut: ``mod + ` `` (in addition to already existing `mod+shift+s`
5. "Go to unit #", mapped to `mod+shift+u` (note that this one was there before, but no longer currently works due to some regression)

**Comment by julen** [comment]
> "Pretranslate using MT", mapped to <kbd>esc enter</kbd>

As noted previous times, there can be multiple MT providers for a source language available at a single time.

> "Reset to current translation", mapped to esc backspace 

Which problem is this solving and how common it is? I feel like one can simply ignore any changes or undo them rather than trying to remember another shortcut, which needs no extra memory from users.

> so remapping this function to <kbd>mod+0</kbd>

<kbd>mod+0</kbd> is used in browsers for resetting the zoom; I'd say this is rather more used than Ubuntu's whatever _internal purpose_ you are referring to.

**Comment by iafan** [comment]
> As noted previous times, there can be multiple MT providers for a source language available at a single time.

Yes, I know. I also suspect this to be a rare case, and I believe that people won't use multiple MT even if the admin would provide several options. So for now I'd say that we need to use the first one. Some time ago in another discussion thread I suggested using a drop-down for MT providers, which I believe is the proper way to handle this. One selects their preferred provider, then clicks the button or uses the shortcut to pretranslate text.

> Which problem is this solving and how common it is?

I would add this for completeness: in the future I'd like to add the way to go through the history (timeline) to revert to some previous translations, so there needs to be a way to reset this without having to do reloads or going one unit down, cancelling the popup, then going one unit up.

I personally sometimes translate something, then change my mind, then translate again. It's easier to press this shortcut than to make edits on top of existing ones.

> mod+0 is used in browsers for resetting the zoom;

Yep, my bad. Time to have some sleep, I guess. `mod+shift+0` is better.

**Comment by iafan** [comment]
> Ubuntu's whatever internal purpose you are referring to.

For curious minds, Ubuntu uses this to start IME input.

**Comment by julen** [comment]
>> As noted previous times, there can be multiple MT providers for a source language available at a single time.
> 
> Yes, I know. I also suspect this to be a rare case, and I believe that people won't use multiple MT even if the admin would provide several options.

I don't believe it's that uncommon. Having Google and Bing enabled already exposes this situation for many major locales, as well as many other minor ones. It's the case on MLO for example. I know people would usually decide on one and stick to that (personally I don't use the feature even if available), and I agree with your other suggestion to have a preferred one and simplify the situation with respect to shortcuts.

> I would add this for completeness: in the future I'd like to add the way to go through the history (timeline) to revert to some previous translations, so there needs to be a way to reset this without having to do reloads or going one unit down, cancelling the popup, then going one unit up.

Maybe I misunderstood but opening the timeline, clicking on a previous unit and submitting doesn't require reloads or cancelling popups.

> I personally sometimes translate something, then change my mind, then translate again. It's easier to press this shortcut than to make edits on top of existing ones.

I'm not sure I follow here. Are you referring to undoing the last submission action for, say unit `uid = N`, when you are in unit `uid != N`?

**Comment by iafan** [comment]
> opening the timeline, clicking on a previous unit and submitting doesn't require reloads or cancelling popups

But it requires using the mouse.

> Are you referring to undoing the last submission action for, say unit uid = N, when you are in unit uid != N?

No, I'm referring to resetting unsubmitted changes and going back to the current translation (as if I reloaded the unit).

**Comment by julen** [comment]
>> opening the timeline, clicking on a previous unit and submitting doesn't require reloads or cancelling popups
> 
> But it requires using the mouse.

Unless I'm failing to grasp the point, I don't see relevance between this and _resetting to current translation_. Honestly, I don't believe there's anything wrong with <kbd>mod+z</kbd> —it's universally known—, and throughout all years that I've been using Pootle as a localizer such a need to revert to the initial state has been perfectly covered by it.

**Comment by iafan** [comment]
> I don't believe there's anything wrong with mod+z

Nothing wrong per se. Undo has potentially multiple steps, while I'd like to have just one.

The `esc backspace` shortcut doesn't seem to intefere with anything, is trivial to implement, and I believe will have more tangible justification to everyone else, not just me, as we implement the abovementioned way to pick translations from timeline, and with implementing `esc space` and `esc enter` (to e.g. quickly compare existing and MT translation). We can postpone implementing it till all other stuff happens.
